pkg/apis/seedmanagement/types_managedseed.go

```diff
@@ -0,0 +1,182 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package seedmanagement
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardenletconfigv1alpha1 "github.com/gardener/gardener/pkg/gardenlet/apis/config/v1alpha1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ManagedSeed represents a Shoot that is registered as Seed.
+type ManagedSeed struct {
+	metav1.TypeMeta
+	// Standard object metadata.
+	metav1.ObjectMeta
+	// Specification of the ManagedSeed.
+	Spec ManagedSeedSpec
+	// Most recently observed status of the ManagedSeed.
+	Status ManagedSeedStatus
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ManagedSeedList is a list of ManagedSeed objects.
+type ManagedSeedList struct {
+	metav1.TypeMeta
+	// Standard list object metadata.
+	metav1.ListMeta
+	// Items is the list of ManagedSeeds.
+	Items []ManagedSeed
+}
+
+// ManagedSeedSpec is the specification of a ManagedSeed.
+type ManagedSeedSpec struct {
+	// Shoot is the Shoot that will be registered as Seed.
+	Shoot *Shoot
+	// Seed describes the Seed that will be registered.
+	// Either Seed or Gardenlet must be specified.
+	Seed *SeedTemplateSpec
+	// Gardenlet specifies gardenlet deployment parameters and the GardenletConfiguration used to configure gardenlet.
+	Gardenlet *Gardenlet
+}
+
+// Shoot identifies the Shoot that will be registered as Seed.
+type Shoot struct {
+	// Name is the name of the Shoot that will be registered as Seed.
+	Name string
+	// APIServer specifies certain kube-apiserver parameters of the Shoot that will be registered as Seed.
+	APIServer *APIServer
+}
+
+// SeedTemplateSpec describes the data a Seed should have when created from a template.
+type SeedTemplateSpec struct {
+	// Standard object metadata.
+	metav1.ObjectMeta
+	// Specification of the desired behavior of the Seed.
+	Spec gardencorev1beta1.SeedSpec
+}
+
+// Gardenlet specifies gardenlet deployment parameters and the GardenletConfiguration used to configure gardenlet.
+type Gardenlet struct {
+	// Deployment specifies certain gardenlet deployment parameters, such as the number of replicas,
+	// the image, which bootstrap mechanism to use (bootstrap token / service account), etc.
+	Deployment *GardenletDeployment
+	// Config is the GardenletConfiguration used to configure gardenlet.
+	Config *gardenletconfigv1alpha1.GardenletConfiguration
+	// Bootstrap is the mechanism that should be used for bootstrapping gardenlet connection to the Garden cluster. One of ServiceAccount, Token.
```

Comment:
`GardenConnectionBootstrap` is better in my opinion, because `Bootstrap` is too generic, and at least for me was quite confusing what "bootstrap" meant until I took a closer look at the code. Also, there is a naming parallel with the `SeedConnection` option that would be lost if this one is renamed.

Reply:
Not sure why you comment this :) My suggestion above is just correcting the doc string. I'm fine with the name itself.

Reply:
Oh, sorry, my mistake ...